packages/store/src/internal/state-factory.ts

```diff
@@ -84,17 +84,22 @@ export class StateFactory {
     return value;
   }
 
+  private static checkStatesAreValid(stateClasses: StateClass[]): void {
+    stateClasses.forEach(StoreValidators.getValidStateMeta);
+  }
+
   /**
    * Add a new state to the global defs.
    */
   add(stateClasses: StateClass[]): MappedStore[] {
-    this.checkStatesAreValid(stateClasses);
-    this.checkForDuplicateStateNames(stateClasses);
+    StateFactory.checkStatesAreValid(stateClasses);
+    const { newStates } = this.addToStatesMap(stateClasses);
+    if (!newStates.length) return [];
```

Comment:
I stick to one rule that makes it easier to debug
`1 method = 1 return`

Reply:
This is a guard clause that reduces the nesting within the method.
It is a tradeoff between the two, and reduced nesting is the better one to opt for in my opinion.
(we have had this conversation before... guard clauses are the one exception to the 1 mthod 1 return rule from Code Complete)